07_deploy_masters.sh

```diff
@@ -37,4 +37,41 @@ done
 for node in $(jq -r .nodes[].name $MASTER_NODES_FILE); do
   openstack baremetal node deploy --config-drive configdrive $node
 done
-# FIXME(shardy) we should wait for the node deploy to complete (or fail)
+
+# Note we have to tolerate failure with the || echo "0" due to this issue
+# https://storyboard.openstack.org/#!/story/2005093
+NUM_ACTIVE=$(openstack baremetal node list --fields name --fields provision_state | grep master | grep active | wc -l || echo 0)
+while [ "$NUM_ACTIVE" -ne 3 ]; do
+  if openstack baremetal node list --fields name --fields provision_state | grep master | grep error; then
+    openstack baremetal node list
+    echo "Error detected waiting for baremetal nodes to become active" >&2
+    exit 1
+  fi
+  sleep 10
+  NUM_ACTIVE=$(openstack baremetal node list --fields name --fields provision_state | grep master | grep active | wc -l || echo 0)
+done
+
+echo "Master nodes active"
+openstack baremetal node list
+
+NUM_LEASES=$(sudo virsh net-dhcp-leases baremetal | grep master | wc -l)
+while [ "$NUM_LEASES" -ne 3 ]; do
+  sleep 10
+  NUM_LEASES=$(sudo virsh net-dhcp-leases baremetal | grep master | wc -l)
+done
+
+echo "Master nodes up, you can ssh to the following IPs with core@<IP>"
+sudo virsh net-dhcp-leases baremetal
+
+while ! oc --config ocp/auth/kubeconfig get nodes | grep -q master; do
+  echo "Waiting for the master API to become ready..."
+  sleep 10
+done
+
+NODES_ACTIVE=$(oc --config ocp/auth/kubeconfig get nodes | grep master | grep Ready | wc -l)
```

Comment:
At this point you might also get nodes in a "NotReady" state
master-0   NotReady   master    26s       v1.12.4+50c2f2340a

After about a minute they seem to switch to "Ready", but the VIP still hasn't moved to a master node, this appears to happen (in local tests) about 4 minutes later, does this matter? If so we also should test for this. Something like
[[ $(ssh -o StrictHostKeyChecking=no  core@api.ostest.test.metalkube.org hostname) =~ master- ]] || echo Still Waiting

Reply:
Ah thanks, I'd missed that, sure we can add the hostname check, or make the grep stricter I guess?

Reply:
I think I'd go for the hostname check as that seems to be the last thing that happens